packages/liveblocks-react-ui/src/primitives/Composer/index.tsx

```diff
@@ -1027,6 +1031,17 @@ const ComposerForm = forwardRef<HTMLFormElement, ComposerFormProps>(
     const ref = useRef<HTMLFormElement>(null);
     const mergedRefs = useRefs(forwardedRef, ref);
     const fileInputRef = useRef<HTMLInputElement>(null);
+    const syncSource = useSyncSource();
+
+    // Mark the composer as a pending update when it isn't empty
+    // or when attachments are being uploaded
+    const isPending = !preventUnsavedChanges
+      ? false
+      : !isEmpty || isUploadingAttachments || attachments.length > 0;
```

Comment:
@marcbouchenoire One extra change I made here is that I also added the `attachments.length > 0` check here, meaning even if the upload has finished successfully, but the comment has not been submitted yet, it will still be considered "pending".

Reply:
Please let me know if that is wrong in any way!

Reply:
Makes sense! Maybe we could update the various comments that mention attachments uploading explicitly since that's no longer the condition.